### PR TITLE
Question Functionality working. 

### PR DESCRIPTION
**Pull request by SBrandon0432**
image
image

### CODE REVIEW

**Review by SBrandon0432 — COMMENTED** [review]
good changes, needed to commit a little more on the changes on each function update

File: src/Components/QuestionList/QuestionCard/QuestionCard.tsx [hunk 1]
```diff
@@ -4,36 +4,68 @@ import Badge from "react-bootstrap/Badge";
...
+}: any) => {
+    return (
+        <>
+            {currentQuestions.map((question: any, index: number) => {
+                let qnsweredColor = question.correct ? "success" : "danger";
+                const difficultyColor = {
+                    easy: "#198754", //success
+                    medium: "#ffc107", // warning
+                    hard: "#dc3545", // danger
+                } as any;
```
Comment (CommanderHamster) [inline src/Components/QuestionList/QuestionCard/QuestionCard.tsx:20]: Change this to an enum and put into a utils or types file.
```
enum DifficultyColor {
  Easy: "#198754",
  ...
}
```

File: src/Components/QuestionList/QuestionCard/QuestionCard.tsx [hunk 2]
```diff
@@ -4,36 +4,68 @@ import Badge from "react-bootstrap/Badge";
...
-        );
-    });
+export const QuestionCard = ({
+    currentQuestions,
+    setCurrentQuestion,
+    setShowQuestionList,
+}: any) => {
+    return (
+        <>
+            {currentQuestions.map((question: any, index: number) => {
```
Comment (CommanderHamster) [inline src/Components/QuestionList/QuestionCard/QuestionCard.tsx:14]: Question has a type, use it here

File: src/Components/QuestionList/QuestionCard/QuestionCard.tsx [hunk 3]
```diff
@@ -4,36 +4,68 @@ import Badge from "react-bootstrap/Badge";
...
-                    </Card.Body>
-                </Card>
-            </Button>
-        );
-    });
+export const QuestionCard = ({
+    currentQuestions,
+    setCurrentQuestion,
+    setShowQuestionList,
+}: any) => {
```
Comment (CommanderHamster) [inline src/Components/QuestionList/QuestionCard/QuestionCard.tsx:11]: Define your props

File: src/Components/QuestionList/QuestionCard/QuestionCard.tsx [hunk 4]
```diff
@@ -4,36 +4,68 @@ import Badge from "react-bootstrap/Badge";
...
+                const difficultyColor = {
+                    easy: "#198754", //success
+                    medium: "#ffc107", // warning
+                    hard: "#dc3545", // danger
+                } as any;
+                return (
+                    <Button
+                        variant="outline-light"
+                        size="lg"
+                        style={{ color: "black" }}
```
Comment (CommanderHamster) [inline src/Components/QuestionList/QuestionCard/QuestionCard.tsx:25]: Don't override style here

File: src/Components/QuestionList/QuestionCard/QuestionCard.tsx [hunk 5]
```diff
@@ -4,36 +4,68 @@ import Badge from "react-bootstrap/Badge";
...
+                                <Card.Text>{question.answer}</Card.Text>
+                                <span
+                                    className="difficulty"
+                                    style={{
+                                        borderTop: `4rem solid ${
+                                            difficultyColor[
+                                                question.difficulty[0]
+                                            ]
+                                        }`,
+                                    }}
```
Comment (CommanderHamster) [inline src/Components/QuestionList/QuestionCard/QuestionCard.tsx:43]: You can make difficulty classNames and then apply those to your components and then do your CSS in the CSS file where it belongs

File: src/Components/QuestionList/QuestionCard/QuestionCard.tsx [hunk 6]
```diff
@@ -4,36 +4,68 @@ import Badge from "react-bootstrap/Badge";
...
-    });
+export const QuestionCard = ({
+    currentQuestions,
+    setCurrentQuestion,
+    setShowQuestionList,
+}: any) => {
+    return (
+        <>
+            {currentQuestions.map((question: any, index: number) => {
+                let qnsweredColor = question.correct ? "success" : "danger";
```
Comment (CommanderHamster) [inline src/Components/QuestionList/QuestionCard/QuestionCard.tsx:15]: questionColor

Comment (CommanderHamster) [reply src/Components/QuestionList/QuestionCard/QuestionCard.tsx:15]: Would rather see this as `questionVariant` and then have it be "outline-success" or "outline-danger"

File: src/Components/QuestionList/QuestionCard/QuestionCard.tsx [hunk 7]
```diff
@@ -4,36 +4,68 @@ import Badge from "react-bootstrap/Badge";
...
+                                    {question.tags.map(
+                                        (tag: string, index: number) => {
+                                            return (
+                                                <Badge bg="dark" key={index}>
+                                                    {tag.toLocaleUpperCase()}
+                                                </Badge>
+                                            );
+                                        }
+                                    )}
+                                </div>
```
Comment (CommanderHamster) [inline src/Components/QuestionList/QuestionCard/QuestionCard.tsx:63]: Make this its own component

File: src/Components/QuestionList/QuestionCard/QuestionCard.tsx [hunk 8]
```diff
@@ -4,36 +4,68 @@ import Badge from "react-bootstrap/Badge";
...
+                                <div>
+                                    {question.tags.map(
+                                        (tag: string, index: number) => {
+                                            return (
+                                                <Badge bg="dark" key={index}>
+                                                    {tag.toLocaleUpperCase()}
+                                                </Badge>
+                                            );
+                                        }
+                                    )}
```
Comment (CommanderHamster) [inline src/Components/QuestionList/QuestionCard/QuestionCard.tsx:62]: Make this its own component

File: src/Components/QuestionList/QuestionCard/QuestionCardS.scss [hunk 9]
```diff
@@ -24,7 +24,7 @@
         left: 0;
         width: 1rem;
         height: 1rem;
-        border-top: 4rem solid $warning;
+        // border-top: 4rem solid $warning;
```
Comment (CommanderHamster) [inline src/Components/QuestionList/QuestionCard/QuestionCardS.scss:27]: remove

**Pull request merged**